src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp

```diff
@@ -207,12 +207,45 @@ void GraphingCalculator::OnEquationsVectorChanged(IObservableVector<EquationView
     GraphingControl->PlotGraph(false);
 }
 
+wstringstream GraphingCalculator::FormatTraceValue(double min, double max, float pointValue)
+{
+    wstringstream traceValueString;
+
+    // Extract precision we will round to 
+    auto precision = static_cast<int>(floor(log10(max - min)) - 3);
+
+    // Determine if we want to show scientific notation instead
+    if (precision <= -7 || precision  >= 7)
+    {
+        traceValueString << scientific;
+    }
+    else
+    {
+        traceValueString << fixed;
+    }
+
+
+    // If we are rounding to a decimal place, set the precision
+    if (precision < 0)
+    {
+        traceValueString << setprecision(::min(7, abs(precision))) << pointValue;
```

Comment:
we can replace the conditional with `setprecision(::max(0, ::min(7, precision))`

Reply:
As discussed, it will be simpler to leave the logic as is since otherwise we will need to inverse the value of precision